src/consumers/consumers.h

```diff
@@ -64,6 +37,56 @@ void* consume_datagram(void* args);
  */
 void consumer_request_output();
 
+/**
+ * Processes datagram struct into metric 
+ * @arg m - Metrics struct acting as metrics wrapper
+ * @arg datagram - Datagram to be processed
+ */
+void process_datagram(metrics* m, statsd_datagram* datagram);
+
+/**
+ * Frees metric
+ * @arg metric - Metric to be freed
+ */
+void free_metric(metric* metric);
+
+/**
+ * Writes information about recorded metrics into file
+ * @arg m - Metrics struct (what values to print)
+ * @arg config - Config containing information about where to output
+ */
+void print_metrics(metrics* m, agent_config* config);
+
+/**
+ * Finds metric by name
+ * @arg name - Metric name to search for
+ * @arg out - Placeholder metric
+ * @return 1 when any found
+ */
+int find_metric_by_name(metrics* m, char* name, metric** out);
+
+/**
+ * Creates metric
+ * @arg datagram - Datagram with data that should populate new metric
+ * @arg out - Placeholder metric
+ * @return 1 on success
+ */
+int create_metric(statsd_datagram* datagram, metric** out);
+
+/**
+ * Adds metric to hashtable
+ * @arg counter - Metric to be saved
+ */
+void save_metric(metrics* m, char* key, metric* item);
```

Comment:
Word "save" sounds like storing something on non-volatile place, I'd like "add" more but it's just matter of preference.